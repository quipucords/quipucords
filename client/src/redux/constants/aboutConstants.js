const ABOUT_DIALOG_OPEN = 'ABOUT_DIALOG_OPEN';
const ABOUT_DIALOG_CLOSE = 'ABOUT_DIALOG_CLOSE';

export { ABOUT_DIALOG_OPEN, ABOUT_DIALOG_CLOSE };
