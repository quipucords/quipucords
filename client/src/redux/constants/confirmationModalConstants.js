const CONFIRMATION_MODAL_SHOW = 'CONFIRMATION_MODAL_SHOW';
const CONFIRMATION_MODAL_HIDE = 'CONFIRMATION_MODAL_HIDE';

export { CONFIRMATION_MODAL_SHOW, CONFIRMATION_MODAL_HIDE };
