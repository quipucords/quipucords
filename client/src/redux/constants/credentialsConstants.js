
const LOAD_CREDENTIALS_SUCCESS = 'LOAD_CREDENTIALS_SUCCESS';

export {
  LOAD_CREDENTIALS_SUCCESS
}
