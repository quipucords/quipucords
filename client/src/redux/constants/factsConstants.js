const ADD_FACTS_ERROR = 'ADD_FACTS_ERROR';
const ADD_FACTS_LOADING = 'ADD_FACTS_LOADING';
const ADD_FACTS_SUCCESS = 'ADD_FACTS_SUCCESS';

export { ADD_FACTS_ERROR, ADD_FACTS_LOADING, ADD_FACTS_SUCCESS };
