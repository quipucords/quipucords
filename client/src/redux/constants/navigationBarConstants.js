const NAV_TOGGLE_COLLAPSE = 'NAV_TOGGLE_COLLAPSE';
const NAV_COLLAPSE = 'NAV_COLLAPSE';
const NAV_OPEN = 'NAV_OPEN';

export { NAV_TOGGLE_COLLAPSE, NAV_COLLAPSE, NAV_OPEN };
