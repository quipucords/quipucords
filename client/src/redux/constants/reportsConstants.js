const GET_REPORT = 'GET_REPORT';
const GET_MERGE_REPORT = 'GET_MERGE_REPORT';

export { GET_REPORT, GET_MERGE_REPORT };
