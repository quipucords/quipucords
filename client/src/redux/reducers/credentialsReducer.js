import _ from 'lodash';
import helpers from '../../common/helpers';
import { credentialsTypes } from '../constants';

const initialState = {
  view: {
    error: false,
    errorMessage: '',
    pending: false,
    fulfilled: false,
    credentials: []
  },

  update: {
    error: false,
    errorMessage: '',
    pending: false,
    fulfilled: false,
    credential: null,
    credentialType: '',
    show: false,
    add: false,
    edit: false,
    delete: false
  }
};

const credentialsReducer = function(state = initialState, action) {
  switch (action.type) {
    // Show/Hide
    case credentialsTypes.CREATE_CREDENTIAL_SHOW:
      return helpers.setStateProp(
        'update',
        {
          show: true,
          add: true,
          credentialType: action.credentialType
        },
        {
          state,
          initialState
        }
      );

    case credentialsTypes.EDIT_CREDENTIAL_SHOW:
      return helpers.setStateProp(
        'update',
        {
          show: true,
          edit: true,
          credential: action.credential
        },
        {
          state,
          initialState
        }
      );

    case credentialsTypes.UPDATE_CREDENTIAL_HIDE:
      return helpers.setStateProp(
        'update',
        {
          show: false
        },
        {
          state,
          initialState
        }
      );

    // Error/Rejected
    case helpers.REJECTED_ACTION(credentialsTypes.ADD_CREDENTIAL):
      return helpers.setStateProp(
        'update',
        {
          error: action.error,
          errorMessage: helpers.getErrorMessageFromResults(action.payload),
          pending: false,
          add: true
        },
        {
          state,
          reset: false
        }
      );

    case helpers.REJECTED_ACTION(credentialsTypes.DELETE_CREDENTIAL):
    case helpers.REJECTED_ACTION(credentialsTypes.DELETE_CREDENTIALS):
      return helpers.setStateProp(
        'update',
        {
          error: action.error,
          errorMessage: helpers.getErrorMessageFromResults(action.payload),
          delete: true,
          pending: false
        },
        {
          state,
          reset: false
        }
      );

    case helpers.REJECTED_ACTION(credentialsTypes.UPDATE_CREDENTIAL):
      return helpers.setStateProp(
        'update',
        {
          error: action.error,
          errorMessage: helpers.getErrorMessageFromResults(action.payload),
          pending: false,
          edit: true
        },
        {
          state,
          reset: false
        }
      );

    case helpers.REJECTED_ACTION(credentialsTypes.GET_CREDENTIAL):
    case helpers.REJECTED_ACTION(credentialsTypes.GET_CREDENTIALS):
      return helpers.setStateProp(
        'view',
        {
          error: action.error,
          errorMessage: helpers.getErrorMessageFromResults(action.payload)
        },
        {
          state,
          initialState
        }
      );

    // Loading/Pending
    case helpers.PENDING_ACTION(credentialsTypes.ADD_CREDENTIAL):
      return helpers.setStateProp(
        'update',
        {
          pending: true,
          error: false,
          fulfilled: false
        },
        {
          state,
          reset: false
        }
      );

    case helpers.PENDING_ACTION(credentialsTypes.DELETE_CREDENTIAL):
    case helpers.PENDING_ACTION(credentialsTypes.DELETE_CREDENTIALS):
      return helpers.setStateProp(
        'update',
        {
          pending: true,
          delete: true,
          error: false,
          fulfilled: false
        },
        {
          state,
          reset: false
        }
      );

    case helpers.PENDING_ACTION(credentialsTypes.UPDATE_CREDENTIAL):
      return helpers.setStateProp(
        'update',
        {
          pending: true,
          error: false,
          fulfilled: false
        },
        {
          state,
          reset: false
        }
      );

    case helpers.PENDING_ACTION(credentialsTypes.GET_CREDENTIAL):
    case helpers.PENDING_ACTION(credentialsTypes.GET_CREDENTIALS):
      return helpers.setStateProp(
        'view',
        {
          pending: true,
          credentials: state.view.credentials
        },
        {
          state,
          initialState
        }
      );

    // Success/Fulfilled
    case helpers.FULFILLED_ACTION(credentialsTypes.ADD_CREDENTIAL):
      return helpers.setStateProp(
        'update',
        {
          credential: action.payload,
          fulfilled: true,
          pending: false,
          error: false,
          errorMessage: ''
        },
        {
          state,
          reset: false
        }
      );

    case helpers.FULFILLED_ACTION(credentialsTypes.DELETE_CREDENTIAL):
    case helpers.FULFILLED_ACTION(credentialsTypes.DELETE_CREDENTIALS):
      return helpers.setStateProp(
        'update',
        {
          credential: action.payload,
          fulfilled: true,
          pending: false,
          delete: true,
          error: false,
          errorMessage: ''
        },
        {
          state,
          reset: false
        }
      );

    case helpers.FULFILLED_ACTION(credentialsTypes.UPDATE_CREDENTIAL):
      return helpers.setStateProp(
        'update',
        {
          credential: action.payload,
          fulfilled: true,
          pending: false,
          error: false,
          errorMessage: ''
        },
        {
          state,
          reset: false
        }
      );

    case helpers.FULFILLED_ACTION(credentialsTypes.GET_CREDENTIAL):
    case helpers.FULFILLED_ACTION(credentialsTypes.GET_CREDENTIALS):
      return helpers.setStateProp(
        'view',
        {
          credentials: _.get(action, 'payload.data.results', []),
          fulfilled: true
        },
        {
          state,
          initialState
        }
      );

    case credentialsTypes.UPDATE_CREDENTIAL_RESET_STATUS:
      return helpers.setStateProp(
        'update',
        {
          error: false,
          errorMessage: ''
        },
        {
          state,
          reset: false
        }
      );

    default:
      return state;
  }
};

export { initialState, credentialsReducer };

export default credentialsReducer;
