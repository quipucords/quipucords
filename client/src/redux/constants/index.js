import * as aboutTypes from './aboutConstants';
import * as credentialsTypes from './credentialsConstants';
import * as factsTypes from './factsConstants';
import * as navigationBarTypes from './navigationBarConstants';
import * as reportsTypes from './reportsConstants';
import * as scansTypes from './scansConstants';
import * as sourcesTypes from './sourcesConstants';
import * as viewToolbarTypes from './viewToolbarConstants';

const reduxTypes = {
  about: aboutTypes,
  credentials: credentialsTypes,
  facts: factsTypes,
  navigation: navigationBarTypes,
  reports: reportsTypes,
  scans: scansTypes,
  sources: sourcesTypes,
  viewToolbar: viewToolbarTypes
};

export {
  reduxTypes,
  aboutTypes,
  credentialsTypes,
  factsTypes,
  navigationBarTypes,
  reportsTypes,
  scansTypes,
  sourcesTypes,
  viewToolbarTypes
};

export default reduxTypes;
