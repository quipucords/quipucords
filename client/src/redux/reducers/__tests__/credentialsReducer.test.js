import credentialsReducer from '../credentialsReducer';
import { credentialsTypes as types } from '../../constants';
import helpers from '../../../common/helpers';

describe('CredentialsReducer', () => {
  it('should return the initial state', () => {
    expect(credentialsReducer.initialState).toBeDefined();
  });

  it('should handle specific defined types', () => {
    const specificTypes = [types.CREATE_CREDENTIAL_SHOW, types.EDIT_CREDENTIAL_SHOW, types.UPDATE_CREDENTIAL_HIDE];

    specificTypes.forEach(value => {
      const dispatched = {
        type: value
      };

      const resultState = credentialsReducer(undefined, dispatched);

      expect({ type: value, result: resultState }).toMatchSnapshot(`defined type ${value}`);
    });
  });

  it('should handle all defined error types', () => {
    const specificTypes = [
      types.ADD_CREDENTIAL,
      types.DELETE_CREDENTIAL,
      types.DELETE_CREDENTIALS,
      types.UPDATE_CREDENTIAL,
      types.GET_CREDENTIALS
    ];

    specificTypes.forEach(value => {
      const dispatched = {
        type: helpers.REJECTED_ACTION(value),
        error: true,
        payload: {
          message: 'MESSAGE',
          response: {
            status: 0,
            statusText: 'ERROR TEST',
            data: {
              detail: 'ERROR'
            }
          }
        }
      };

      const resultState = credentialsReducer(undefined, dispatched);

      expect({ type: helpers.REJECTED_ACTION(value), result: resultState }).toMatchSnapshot(`rejected types ${value}`);
    });
  });

  it('should handle all defined pending types', () => {
    const specificTypes = [
      types.ADD_CREDENTIAL,
      types.DELETE_CREDENTIAL,
      types.DELETE_CREDENTIALS,
      types.UPDATE_CREDENTIAL,
      types.GET_CREDENTIALS
    ];

    specificTypes.forEach(value => {
      const dispatched = {
        type: helpers.PENDING_ACTION(value)
      };

      const resultState = credentialsReducer(undefined, dispatched);

      expect({ type: helpers.PENDING_ACTION(value), result: resultState }).toMatchSnapshot(`pending types ${value}`);
    });
  });

  it('should handle all defined fulfilled types', () => {
    const specificTypes = [
      types.ADD_CREDENTIAL,
      types.DELETE_CREDENTIAL,
      types.DELETE_CREDENTIALS,
      types.UPDATE_CREDENTIAL,
      types.GET_CREDENTIALS
    ];

    specificTypes.forEach(value => {
      const dispatched = {
        type: helpers.FULFILLED_ACTION(value),
        payload: {
          data: {
            test: 'success'
          }
        }
      };

      const resultState = credentialsReducer(undefined, dispatched);

      expect({ type: helpers.FULFILLED_ACTION(value), result: resultState }).toMatchSnapshot(
        `fulfilled types ${value}`
      );
    });
  });
});
