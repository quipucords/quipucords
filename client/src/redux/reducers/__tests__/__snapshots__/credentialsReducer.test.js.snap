// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`defined type CREATE_CREDENTIAL_SHOW 1`] = `
Object {
  "result": Object {
    "update": Object {
      "add": true,
      "credential": null,
      "credentialType": undefined,
      "delete": false,
      "edit": false,
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
      "show": true,
    },
    "view": Object {
      "credentials": Array [],
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
    },
  },
  "type": "CREATE_CREDENTIAL_SHOW",
}
`;

exports[`defined type EDIT_CREDENTIAL_SHOW 1`] = `
Object {
  "result": Object {
    "update": Object {
      "add": false,
      "credential": undefined,
      "credentialType": "",
      "delete": false,
      "edit": true,
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
      "show": true,
    },
    "view": Object {
      "credentials": Array [],
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
    },
  },
  "type": "EDIT_CREDENTIAL_SHOW",
}
`;

exports[`defined type UPDATE_CREDENTIAL_HIDE 1`] = `
Object {
  "result": Object {
    "update": Object {
      "add": false,
      "credential": null,
      "credentialType": "",
      "delete": false,
      "edit": false,
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
      "show": false,
    },
    "view": Object {
      "credentials": Array [],
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
    },
  },
  "type": "UPDATE_CREDENTIAL_HIDE",
}
`;

exports[`fulfilled types ADD_CREDENTIAL 1`] = `
Object {
  "result": Object {
    "update": Object {
      "add": true,
      "credential": Object {
        "test": "success",
      },
      "credentialType": "",
      "delete": false,
      "edit": false,
      "error": false,
      "errorMessage": "",
      "fulfilled": true,
      "pending": false,
      "show": false,
    },
    "view": Object {
      "credentials": Array [],
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
    },
  },
  "type": "ADD_CREDENTIAL_FULFILLED",
}
`;

exports[`fulfilled types DELETE_CREDENTIAL 1`] = `
Object {
  "result": Object {
    "update": Object {
      "add": false,
      "credential": null,
      "credentialType": "",
      "delete": true,
      "edit": false,
      "error": false,
      "errorMessage": "",
      "fulfilled": true,
      "pending": false,
      "show": false,
    },
    "view": Object {
      "credentials": Array [],
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
    },
  },
  "type": "DELETE_CREDENTIAL_FULFILLED",
}
`;

exports[`fulfilled types DELETE_CREDENTIALS 1`] = `
Object {
  "result": Object {
    "update": Object {
      "add": false,
      "credential": null,
      "credentialType": "",
      "delete": true,
      "edit": false,
      "error": false,
      "errorMessage": "",
      "fulfilled": true,
      "pending": false,
      "show": false,
    },
    "view": Object {
      "credentials": Array [],
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
    },
  },
  "type": "DELETE_CREDENTIALS_FULFILLED",
}
`;

exports[`fulfilled types GET_CREDENTIALS 1`] = `
Object {
  "result": Object {
    "update": Object {
      "add": false,
      "credential": null,
      "credentialType": "",
      "delete": false,
      "edit": false,
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
      "show": false,
    },
    "view": Object {
      "credentials": Array [],
      "error": false,
      "errorMessage": "",
      "fulfilled": true,
      "pending": false,
    },
  },
  "type": "GET_CREDENTIALS_FULFILLED",
}
`;

exports[`fulfilled types UPDATE_CREDENTIAL 1`] = `
Object {
  "result": Object {
    "update": Object {
      "add": false,
      "credential": Object {
        "test": "success",
      },
      "credentialType": "",
      "delete": false,
      "edit": true,
      "error": false,
      "errorMessage": "",
      "fulfilled": true,
      "pending": false,
      "show": false,
    },
    "view": Object {
      "credentials": Array [],
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
    },
  },
  "type": "UPDATE_CREDENTIAL_FULFILLED",
}
`;

exports[`pending types ADD_CREDENTIAL 1`] = `
Object {
  "result": Object {
    "update": Object {
      "add": false,
      "credential": null,
      "credentialType": "",
      "delete": false,
      "edit": false,
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": true,
      "show": false,
    },
    "view": Object {
      "credentials": Array [],
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
    },
  },
  "type": "ADD_CREDENTIAL_PENDING",
}
`;

exports[`pending types DELETE_CREDENTIAL 1`] = `
Object {
  "result": Object {
    "update": Object {
      "add": false,
      "credential": null,
      "credentialType": "",
      "delete": true,
      "edit": false,
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": true,
      "show": false,
    },
    "view": Object {
      "credentials": Array [],
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
    },
  },
  "type": "DELETE_CREDENTIAL_PENDING",
}
`;

exports[`pending types DELETE_CREDENTIALS 1`] = `
Object {
  "result": Object {
    "update": Object {
      "add": false,
      "credential": null,
      "credentialType": "",
      "delete": true,
      "edit": false,
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": true,
      "show": false,
    },
    "view": Object {
      "credentials": Array [],
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
    },
  },
  "type": "DELETE_CREDENTIALS_PENDING",
}
`;

exports[`pending types GET_CREDENTIALS 1`] = `
Object {
  "result": Object {
    "update": Object {
      "add": false,
      "credential": null,
      "credentialType": "",
      "delete": false,
      "edit": false,
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
      "show": false,
    },
    "view": Object {
      "credentials": Array [],
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": true,
    },
  },
  "type": "GET_CREDENTIALS_PENDING",
}
`;

exports[`pending types UPDATE_CREDENTIAL 1`] = `
Object {
  "result": Object {
    "update": Object {
      "add": false,
      "credential": null,
      "credentialType": "",
      "delete": false,
      "edit": false,
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": true,
      "show": false,
    },
    "view": Object {
      "credentials": Array [],
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
    },
  },
  "type": "UPDATE_CREDENTIAL_PENDING",
}
`;

exports[`rejected types ADD_CREDENTIAL 1`] = `
Object {
  "result": Object {
    "update": Object {
      "add": true,
      "credential": null,
      "credentialType": "",
      "delete": false,
      "edit": false,
      "error": true,
      "errorMessage": "ERROR",
      "fulfilled": false,
      "pending": false,
      "show": false,
    },
    "view": Object {
      "credentials": Array [],
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
    },
  },
  "type": "ADD_CREDENTIAL_REJECTED",
}
`;

exports[`rejected types DELETE_CREDENTIAL 1`] = `
Object {
  "result": Object {
    "update": Object {
      "add": false,
      "credential": null,
      "credentialType": "",
      "delete": true,
      "edit": false,
      "error": true,
      "errorMessage": "ERROR",
      "fulfilled": false,
      "pending": false,
      "show": false,
    },
    "view": Object {
      "credentials": Array [],
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
    },
  },
  "type": "DELETE_CREDENTIAL_REJECTED",
}
`;

exports[`rejected types DELETE_CREDENTIALS 1`] = `
Object {
  "result": Object {
    "update": Object {
      "add": false,
      "credential": null,
      "credentialType": "",
      "delete": true,
      "edit": false,
      "error": true,
      "errorMessage": "ERROR",
      "fulfilled": false,
      "pending": false,
      "show": false,
    },
    "view": Object {
      "credentials": Array [],
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
    },
  },
  "type": "DELETE_CREDENTIALS_REJECTED",
}
`;

exports[`rejected types GET_CREDENTIALS 1`] = `
Object {
  "result": Object {
    "update": Object {
      "add": false,
      "credential": null,
      "credentialType": "",
      "delete": false,
      "edit": false,
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
      "show": false,
    },
    "view": Object {
      "credentials": Array [],
      "error": true,
      "errorMessage": "ERROR",
      "fulfilled": false,
      "pending": false,
    },
  },
  "type": "GET_CREDENTIALS_REJECTED",
}
`;

exports[`rejected types UPDATE_CREDENTIAL 1`] = `
Object {
  "result": Object {
    "update": Object {
      "add": false,
      "credential": null,
      "credentialType": "",
      "delete": false,
      "edit": true,
      "error": true,
      "errorMessage": "ERROR",
      "fulfilled": false,
      "pending": false,
      "show": false,
    },
    "view": Object {
      "credentials": Array [],
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
    },
  },
  "type": "UPDATE_CREDENTIAL_REJECTED",
}
`;
